src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1590,7 +1552,7 @@ def _resolve_service_principal(client, identifier):  # Uses service principal gr
         return result[0].object_id
     if utils.is_guid(identifier):
         return identifier  # assume an object id
-    error = CLIError("Service principal '{}' doesn't exist".format(identifier))
+    error = ResourceNotFoundError("Service principal '{}' doesn't exist".format(identifier))
```

Comment:
What is the use case of the function _resolve_service_principal?

Reply:
It isnt used now.

Reply:
Will remove